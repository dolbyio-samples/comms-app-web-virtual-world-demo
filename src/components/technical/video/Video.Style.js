import styled from "styled-components";

const style = (Component) => styled(Component)`
  object-fit: cover;
`;

export default style;
