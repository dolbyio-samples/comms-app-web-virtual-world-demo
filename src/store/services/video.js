import VoxeetSDK from "@voxeet/voxeet-web-sdk";

export const startVideo = (participant) => {
  return VoxeetSDK.conference
    .startVideo(participant, {
      audio: true,
      video: true,
    })
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};

export const stopVideo = (participant) => {
  return VoxeetSDK.conference
    .stopVideo(participant)
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};
