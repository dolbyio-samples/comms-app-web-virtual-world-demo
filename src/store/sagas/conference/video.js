import { call, select } from "redux-saga/effects";

import { getEntity } from "../../reducers/entities/selectors";
import { startVideo as start, stopVideo as stop } from "../../services/video";
import { displayMessage } from "./message";

export const getParticipant = (state, id) =>
  getEntity(state, "participants", id);

export function* startVideo({ payload }) {
  const participant = yield select(getParticipant, payload);
  if (!participant) return;

  const { error } = yield call(start, participant);
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not start video",
    });
  }
}

export function* stopVideo({ payload }) {
  const participant = yield select(getParticipant, payload);
  if (!participant) return;

  const { error } = yield call(stop, participant);
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not stop video",
    });
  }
}
