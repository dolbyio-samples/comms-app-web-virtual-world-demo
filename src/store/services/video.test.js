import { describe, it, expect, vi } from "vitest";
import { startVideo, stopVideo } from "./video";

const mockStartVideo = vi.fn();
const mockStopVideo = vi.fn();

vi.mock("@voxeet/voxeet-web-sdk", () => ({
  default: {
    get conference() {
      return { startVideo: mockStartVideo, stopVideo: mockStopVideo };
    },
  },
}));

describe("video services test suite", () => {
  it(" should succeed", async () => {
    // arrange
    const participant = { name: "toto" };
    const constraints = { audio: true, video: true };
    mockStartVideo.mockResolvedValue({ response: {} });

    // act
    const { response, error } = await startVideo(participant, constraints);

    // assert
    expect(mockStartVideo).toHaveBeenCalled();
    expect(mockStartVideo).toHaveBeenCalledWith(participant, constraints);
    expect(response).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should fail and return an error", async () => {
    // arrange
    const participant = { name: "toto" };
    const constraints = { video: true, audio: true };
    mockStartVideo.mockRejectedValueOnce(new Error("error"));

    // act
    const { error, response } = await startVideo(participant, constraints);

    // assert
    expect(error.message).toEqual("error");
    expect(response).toBe(undefined);
    expect(mockStartVideo).toHaveBeenCalled();
    expect(mockStartVideo).toHaveBeenCalledWith(participant, constraints);
  });

  it("should succeed", async () => {
    // arrange
    const participant = { name: "toto" };
    mockStopVideo.mockResolvedValue({ response: {} });

    // act
    const { response, error } = await stopVideo(participant);

    // assert
    expect(mockStopVideo).toHaveBeenCalled();
    expect(mockStopVideo).toHaveBeenCalledWith(participant);
    expect(response).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should fail and return an error", async () => {
    // arrange
    const participant = { name: "toto" };
    mockStopVideo.mockRejectedValueOnce(new Error("error"));

    // act
    const { error, response } = await stopVideo(participant);

    // assert
    expect(error).not.toBe(null);
    expect(error.message).toEqual("error");
    expect(error).toMatchSnapshot();
    expect(response).toBe(undefined);
    expect(mockStopVideo).toHaveBeenCalled();
    expect(mockStopVideo).toHaveBeenCalledWith(participant);
  });
});
