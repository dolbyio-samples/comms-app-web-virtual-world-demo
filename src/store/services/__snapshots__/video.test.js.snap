// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`video services test suite >  should succeed 1`] = `
{
  "response": {},
}
`;

exports[`video services test suite > should fail and return an error 1`] = `[Error: error]`;

exports[`video services test suite > should succeed 1`] = `
{
  "response": {},
}
`;
